cache/cache.go

```diff
@@ -72,10 +72,16 @@ func (c *Cache) GetType() string {
 	return CacheType
 }
 
-// getCacheKey returns the cache key for the given key object by computing a
-// checksum of key struct
+// getCacheKey returns the cache key for the given key object by returning
+// the key if type is string or by computing a checksum of key structure
+// if its type is other than string
 func (c *Cache) getCacheKey(key interface{}) string {
-	return strings.ToLower(checksum(key))
+	switch key.(type) {
+	case string:
+		return strings.ToLower(key.(string))
```

Comment:
Can we please not lowercase in case of string type?

Cache key could be case sensitive. Also, different systems could be writing/reading, say to redis like cache, using different libraries, and hence making a decision like this to lowercase cache key is bug prone.

If yes, will file a pr requesting to release new version.

Reply:
Hi @jitendra-1217, you're right, we don't need to lowercase key